installer/templates/phx_web/components.ex

```diff
@@ -270,11 +271,14 @@ defmodule <%= @web_namespace %>.Components do
   def input(%{type: "checkbox"} = assigns) do
     ~H"""
     <label phx-feedback-for={@name} class="flex items-center gap-4 text-sm leading-6 text-zinc-600">
+      <input type="hidden" id={@id || @name} name={@name} value="false" />
       <input
         type="checkbox"
         id={@id || @name}
         name={@name}
+        value="true"
         class="rounded border-zinc-300 text-zinc-900 focus:ring-zinc-900"
+        checked={@rest[:checked]}
```

Comment:
Since we know the possible values for a check box input why not directly use the value for checked attribute?

```
checked={@value || @value =="true"}
```

Reply:
This would indeed work for a boolean field but sometimes custom logic is required to mark the value as checked, using the global @rest lets the developer do it.

I'm upgrading our prod app to LV 0.18 and picking some new stuff from Phoenix as well and select/checkboxes/radios are some of the most annoying fields to work with in some scenarios. So far this is what worked for us.

Reply:
Then I would suggest that it's worth the complexity to have the default be computed. So something like:

```elixir
  defp input_checked(%{checked: checked}, _value) when not is_nil(checked), do: checked
  defp input_checked(_rest, value) when is_boolean(value), do: value
  defp input_checked(_rest, value), do: to_string(value) == "true"

  defp option_selected(%{selected: selected}, _value), do: selected
  defp option_selected(%{value: option_value}, value), do: to_string(option_value) == value


  def input(%{type: "checkbox"} = assigns) do
    ~H"""
      <input type="hidden" id={@id || @name} name={@name} value="false" />
      <input
        type="checkbox"
        id={@id || @name}
        name={@name}
        class="checkbox checkbox-secondary"
        value="true"
        checked={input_checked(@rest, @value)}
      />
      <%= @label %>
    """
  end

  def input(%{type: "select"} = assigns) do
    ~H"""
      <.label for={@id}><%= @label %></.label>
      <select id={@id} name={@name} autocomplete={@name} {@rest}>
        <option
          :for={opt <- @option}
          {assigns_to_attributes(opt, [:selected])}
          selected={option_selected(opt, @value)}
        >
          <%= render_slot(opt) %>
        </option>
      </select>
    """
  end
```

That lets most component users get the correct defaults computed, while people that need to set a value are able to override pretty easily.

Reply:
This looks promising, I may be able to test this snipped with our app this week.

Reply:
Nice! I was able to use this and add a couple extra stuff while testing in our app. I'll update the PR early next week.